server/service/src/report/report_service.rs

```diff
@@ -569,7 +573,14 @@ fn generate_report(
         report.convert_data.clone(),
         &report.convert_data_type,
     )
-    .map_err(ReportError::ConvertDataError)?;
+    .map_err(|err| {
+        error!(
+            "Error transforming data for report {}: {}",
+            report.name,
+            format_error(&err)
+        );
```

Comment:
Adding an error log here, frontend just shows "failed to generate" - good to be able to look at the logs and see what went wrong!

Maybe should add throughout the report generation code but would be bloat, figured adding in the places we touch is an ok start for now?

Reply:
I kind of like this at the top level, either at the entrypoint to the services or even at end consumer (in graphql, when mapping error I guess), although we are often not adding enough info in the error (like the report name you've added, which should technically be in the report error). 

Happy with this addition, it's an improvement